test/integration/quota/quota_test.go

```diff
@@ -88,10 +100,13 @@ func TestQuota(t *testing.T) {
 	controllerCh := make(chan struct{})
 	defer close(controllerCh)
 
-	informers := informers.NewSharedInformerFactory(clientset, controller.NoResyncPeriodFunc())
+	metadataClient := metadata.NewForConfigOrDie(clientConfig)
+	stopOnError := func(error) bool { return true }
```

Comment:
better to have another test case for stopOnError return false

Reply:
I can do this if you think it's the right thing to do, but I just want to point out that we never expect quota and GC controllers to ever be ran with stopOnError to be false. This is an error state and is what causes the bug in #79610 in the first place. I think it would be confusing to have tests for a state we consider to be an error in the first place (even if the controller techinically does still work because the informer never stops).

The only reason we don't change the informer/reflector to always stop on error is because this could break custom CRD controllers that we have no visibility in.

If you do think it's useful to test this state, I can do so here as well as throughout the GC tests, but it will double the number of tests runs in the GC and quota `test/integration` packages because every test runs with metadata informers and so if we do want to test this state we'd probably want to run every test with these states.

Reply:
Same solution as GC test, I've added a check to confirm that the informer is stopped upon CRD uninstall:

https://github.com/kubernetes/kubernetes/pull/97214/files#diff-f9a432a9226aae7f52a9e21d3dbcc354a259a530eba2e2e11c97e0b760f4b2f5R433-R439